lib/Dredd.js

```diff
@@ -1,35 +1,94 @@
 const async = require('async');
-const clone = require('clone');
-const fs = require('fs');
-const path = require('path');
-const request = require('request');
-const url = require('url');
 const parse = require('dredd-transactions/parse');
 const compile = require('dredd-transactions/compile');
 
 const configureReporters = require('./configureReporters');
 const handleRuntimeProblems = require('./handleRuntimeProblems');
 const resolveLocations = require('./resolveLocations');
+const readLocation = require('./readLocation');
+const resolveModule = require('./resolveModule');
 const logger = require('./logger');
-const Runner = require('./TransactionRunner');
+const TransactionRunner = require('./TransactionRunner');
 const { applyConfiguration } = require('./configuration');
-const options = require('./options.json');
 
 
-const PROXY_ENV_VARIABLES = ['HTTP_PROXY', 'HTTPS_PROXY', 'NO_PROXY'];
-const FILE_DOWNLOAD_TIMEOUT = 5000;
+function prefixErrors(decoratedCallback, prefix) {
+  return (error, ...args) => {
+    if (error) { error.message = `${prefix}: ${error.message}`; }
+    decoratedCallback(error, ...args);
+  };
+}
+
+
+function readLocations(locations, options, callback) {
```

Comment:
I feel like readLocations() and parseContent() might get a lot slimmer or even redundant with Promise.all() or Promise.map(), but I decided to stay with callbacks until Dredd drops Node 6 support.